index.bikeshed.bs

```diff
@@ -343,7 +368,65 @@ implementation detail. It is not an API that implementors
 provide to web applications. It is used in this document
 only to simplify the algorithm description.
 
-## Response verification algorithms ## {#verification-algorithms}
+## Request verification algorithms ## {#request-verification-algorithms}
+
+### Opting-in  ### {#opt-in-require-sri-for}
+
+Authors may opt a {{Document}} to require SRI metadata be present for 
+some resource types via a <dfn export>require-sri-for</dfn> <a>Content
+Security Policy</a> directive defined by the following ABNF grammar:
+
+<pre dfn-type="grammar" link-type="grammar">
+    directive-name  = "require-sri-for"
+    directive-value = <a grammar>token</a> *( <a>RWS</a> <a>token</a> )
+</pre>
+
+The following list contains the set of <dfn noexport>known tokens</dfn>:
+
+  * `script` requires SRI for scripts
+  * `style` requires SRI for style sheets
+
+### Parsing `require-sri-for` ### {#parse-require-sri-for}
+
+Given a string (|token list|), this algorithm returns a list of resource
+types which will require integrity checks:
+
+1.  Let the set of |protected resource types| that require SRI be the empty set.
+
+2.  For each |token| in the result of <a lt="split a string on spaces">
+    splitting |token list| on spaces</a> if token matches the grammar
+    for <a>require-sri-for</a>, add |token| to |protected resource types|
+    if |token| is a <a>known token</a>. Otherwise, ignore the token.
+
+3.  Return the set of |protected resource types|.
+
+### Apply |algorithm| to |request| ### {#apply-algorithm-to-request}
+
+This directive’s <a>pre-request check</a> is as follows:
+
+Given a <a>request</a> (|request|) and a <a>policy</a> (|policy|):
+
+1.  Let |protected resource types| be the result of executing
+    [[#parse-require-sri-for]] on this <a>directive</a>'s <a lt="directive value">value</a>.
+
+2.  If |request|'s <a>destination</a> is a <a>ASCII case-insensitive match</a> for at least
+    one token in |protected resource types|, and |request|'s integrity metadata
+    is the empty string:
+
+      1.  Let |violation| be the result of executing <a lt="create a violation object for global">
+          Create a violation object for global, policy, and directive</a> on |document|'s
+          <a>global object</a>, |policy|, and "<a>`require-sri-for`</a>".
+
+      2.  Set |violation|'s <a lt="violation-resource">resource</a> to "`inline`".
```

Comment:
Thanks for raising this!
My initial implementation reports a blocked-uri of "self", which is... not good.
I agree with @mikewest that it would be far more valuable to report the URL of the style/script that hasn't been loaded.